### DISCUSSION TITLE
job_workflow_ref in AWS

### DISCUSSION THREAD

**Discussion by javixeneize** [community/community#30633] [Other Feature Feedback, Questions, & Ideas] [answered]
Hi

Im reading this article, https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/using-openid-connect-with-reusable-workflows, but im unable to make it work in AWS

The article says "Cloud providers that support job_workflow_ref include Google Cloud Platform and HashiCorp Vault". Do you know if aws supports that? If so, do you have any example on how to do it? I have been able to make it work with sub, but not with job_workflow_ref

In aws i have somehting like this

"Action": "sts:AssumeRoleWithWebIdentity",
            "Condition": {
                "ForAnyValue:StringLike": {
                    "token.actions.githubusercontent.com:job_workflow_ref": [
                        "myorg/myrepo/.github/workflows/test.yml@refs/heads/main"
                    ]

Thanks

**Comment by sbkg0002** [comment, marked as answer]
Did you manage to get it to work?

**Reply by javixeneize** [reply]
Yep it works ;)

thanks

**Reply by salvianreynaldi** [reply]
Does this still work?
I have an AWS IAM role, which doesn't work if I add the "token.actions.githubusercontent.com:job_workflow_ref" condition, only the `sub` and `aud` claims work.

[This page](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services) mentions that custom claims don't work in AWS, and [the official action page](https://github.com/aws-actions/configure-aws-credentials/issues/306) confirms the issue.

**Reply by qoomon** [reply]
@javixeneize I'd be interested as well

**Comment by steve-todorov** [comment]
The above "marked as answer" message above is probably a mistake.

Long story short - no, custom claims do not appear to be supported in AWS IAM policies at the time of writing this.
The way you can workaround this is by customizing the `sub` claim at Github to include the workflow in the `sub`:

```
$ gh api -X PUT /orgs/MY_ORG/actions/oidc/customization/sub \
  -f use_default=false \
  -f subject='repo:{{org}}/{{repo}}:ref:{{ref}}:workflow_ref:{{workflow_ref}}' \
  -F include_claim_keys[]=repo \
  -F include_claim_keys[]=ref \
  -F include_claim_keys[]=workflow_ref
```

Then ensure your repository does not use the `default template` for the `sub` claim by executing:
```
$ gh api -X PUT /repos/MY_ORG/MY_REPOSITORY/actions/oidc/customization/sub -F 'use_default=false'
```

The `workflow_ref` will then be part of the `sub` so you'll be able to do  something like this in your IAM policy:
```
StringLike = {
    "token.actions.githubusercontent.com:sub" = [
        "repo:MY_ORG/MY_REPO:ref:refs/heads/*:workflow_ref:MY_ORG/MY_REPO/.github/workflows/my_gh_action_workflow_file.yml@*",
    ]
}
```

The above would allow only a specific workflow in the bucket to be able to assume a role for example.

**Comment by liwadman** [comment]
AWS launched support for job_workflow_ref in winter 2026. Check it out! https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_iam-condition-keys.html#condition-keys-wif

**Reply by steve-todorov** [reply]
Ahh.. that's awesome! When I was setting this up last time we were having issues with it. Good to know it's now fixed and there's no reason to go the extra mile with additional configuration. :)

**Reply by IUADE0H5** [reply]
FYI: that doesn't cover GHE server
